database/db_map.py

```diff
@@ -43,4 +41,74 @@ def get_receive_stations(self, rcv_station_dict):
                                                     categories: list(int),
                                                     items: list(int)}, ...}
         """
-        raise NotImplementedError
+        response = self.db.get("receiver_stations", None)
+        self.checking(rcv_station_dict, response)
+        return response
+
+    def checking(self, user_station, stations):
+        """
+        Return all suitable for a user stations,
+        otherwise return []
+        :param user_station: dict = {(str): int, (str): (str), ...}
+        :param stations:  dict = {(str): list, (str): (int), (str): (str), ...}
+        :return: list = [(dict), ...]
+        """
+        suitable_stations = []
+        for station in stations:
+            if self._data_intersection(user_station['categories'], station['categories']):
+                if self._data_intersection(user_station['city_id'], station['locations']):
+                    if self._data_intersection(user_station['receive_station_type'], station['type_id']):
+
+                        s_dates = self._get_all_dates(station['time_from'], station['time_to'])
+                        u_dates = self._get_all_dates(user_station['time_from'], user_station['time_to'])
+                        if self._data_intersection(s_dates, u_dates):
+                            suitable_stations.append(station)
+
+        return suitable_stations
+
+    @staticmethod
+    def _data_intersection(user_data, database_data):
+        """
+        Return True if user_data has some elements in common
+        with database_data
+        :return: (bool)
+        """
+        if not isinstance(user_data, list):
+            user_data = [user_data]
+        if not isinstance(database_data, list):
+            database_data = [database_data]
+
+        user_set = set(user_data)
+        database_set = set(database_data)
+
+        if user_set.intersection(database_set) != set():
+            return True
+        else:
+            return False
```

Comment:
Set intersection need to be used only for categories. Maybe there is no need in one more additional function.